base/assert.hpp

```diff
@@ -30,33 +30,33 @@ namespace my
     ASSERT_CRASH();
 
 // TODO: Evaluate X only once in CHECK().
-#define CHECK(X, msg) do { if (X) {} else { \
+#define CHECK(X, msg) do { if (unlikely(X)) {} else { \
```

Comment:
Should the logic be reversed (likely)? All `CHECK(false)` in the code are for exceptional/rare run-time fatal cases, so obviously, the most popular case is when X == true (and we do nothing in the if {} clause).

Reply:
Indeed. Nice catch, thanks.